Allow going back and forth to the main menu, and toggle fullscreen with Alt+tab

groud opened the pull request:
Once your back to the main menu, you can reload the level really fast, without problem. :)

aaronfranke commented on project.godot:
```diff
@@ -16,7 +16,7 @@ _global_script_class_icons={
 
 config/name="Godot Third-Person Shooter Demo"
-run/main_scene="res://menu/menu.tscn"
+run/main_scene="res://main/main.tscn"
```
Why? What's the purpose of a scene that immediately loads another?

groud replied:
This allow keeping a global state. If you unload all you nodes using `get_tree().change_scene(...)`, you drop all possibilities to handle some things globally. Typically here, I use it to toggle the fullscreen mode.

aaronfranke commented on project.godot:
```diff
@@ -179,6 +179,11 @@ view_down={
 }
+toggle_fullscreen={
+"deadzone": 0.5,
+"events": [ Object(InputEventKey,"resource_local_to_scene":false,"resource_name":"","device":0,"alt":true,"shift":false,"control":false,"meta":false,"command":false,"pressed":false,"scancode":16777221,"unicode":0,"echo":false,"script":null)
```
Alt+Enter is not really a good choice for this. Few people would know about it, and it causes issues on the menu (pressing enter will start the game). I suggest F11 instead.

Calinou replied:
I suggest adding both <kbd>Alt + Enter</kbd> and <kbd>F11</kbd>, as the former has been a de facto standard for toggling fullscreen in games for decades now. Many recent AAAs still support it :slightly_smiling_face:

groud replied:
Yeah Alt+Enter is pretty common in a lot of games, it's better if we can keep it. I'll probably add the F11 shortcut and eventually an option in the settings.

 I might be able to block the event so that it does not trigger the game starting. If I can't do it, I'll remove the Alt+Enter shortcut.

groud commented:
I pushed new changes:
- both Alt-Enter and F11 work to toggle the fullscreen (the event is set as handled, so it does not trigger the Enter action in the GUI)
- Added an option for fullscreen in the settings menu
- Reorganized the settings buttons using containers
- Moved the GUI styles to a theme
- Made the focus a different color, as it was not visilble before on pressed entries (needed for gamepads)

groud commented:
> The resolution needs to be changed back when going to the menu, or only applied to the 3D rendering.

I don't think we should do that, this would be really annoying to have your window change size every time you go to the menu. 
576 is a very small resolution,

aaronfranke commented:
@groud It doesn't change the window size, only the rendering size. You can set it to 576 and still have it maximized, windowed at any size, or fullscreen.

Removing 576 doesn't work because there is a similar problem with 720.

groud commented:
> The resolution needs to be changed back when going to the menu, or only applied to the 3D rendering.

I did that, changing the UI size was more work. :)

aaronfranke commented:
The exact same issue in the image above still exists. The menu becomes unusable if a low resolution is selected since that resolution is kept when switching back to the menu.

groud commented on menu/menu.gd:
```diff
@@ -3,7 +3,12 @@ extends Spatial
+signal quit # Useless, but needed as there is no clean way to check if a node exposes a signal
+
 func _ready():
+	get_tree().set_screen_stretch(SceneTree.STRETCH_MODE_2D, SceneTree.STRETCH_ASPECT_KEEP, Vector2(1920, 1080))
```
Here it is

groud commented:
> The exact same issue in the image above still exists. The menu becomes unusable if a low resolution is selected since that resolution is kept when switching back to the menu.

You must have done something wrong on your side. It work perfectly for me.
I added the line to switch back to the native resolution at line 11 of menu.gd.

aaronfranke commented:
Thanks!

groud commented:
Nice :) Thanks !
